### DISCUSSION TITLE
Apple M1Pro, cargo run failed, fatal error: 'vpx/vp8.h' file not found

### DISCUSSION THREAD

**Discussion by songwei163** [rustdesk/rustdesk#1535] [Q&A] [answered]
this is my computer:
MacbookPro 14
Apple M1 Pro
When I follow this documentation,https://rustdesk.com/docs/en/dev/build/osx/, I Found some errors when cargo run
Here is details.
error: failed to run custom build command for scrap v0.5.0 (/Users/songwei/work/rustdesk/libs/scrap)

Caused by:
process didn't exit successfully: /Users/songwei/work/rustdesk/target/debug/build/scrap-5b4b8530a1319865/build-script-build (exit status: 101)
--- stdout
cargo:info=x64-osx
cargo:rustc-link-lib=static=yuv
cargo:rustc-link-search=/Users/songwei/vcpkg/installed/x64-osx/lib
cargo:include=/Users/songwei/vcpkg/installed/x64-osx/include
cargo:info=x64-osx
cargo:rustc-link-lib=static=vpx
cargo:rustc-link-search=/Users/songwei/vcpkg/installed/x64-osx/lib
cargo:include=/Users/songwei/vcpkg/installed/x64-osx/include
rerun-if-changed=/Users/songwei/work/rustdesk/libs/scrap/vpx_ffi.h
rerun-if-changed=/Users/songwei/vcpkg/installed/x64-osx/include

--- stderr
/Users/songwei/work/rustdesk/libs/scrap/vpx_ffi.h:1:10: fatal error: 'vpx/vp8.h' file not found
/Users/songwei/work/rustdesk/libs/scrap/vpx_ffi.h:1:10: fatal error: 'vpx/vp8.h' file not found, err: true
thread 'main' panicked at 'called Result::unwrap() on an Err value: ()', libs/scrap/build.rs:128:18
note: run with RUST_BACKTRACE=1 environment variable to display a backtrace

**Comment by sany1231** [comment]
same issu here, did someone soleved this issue?

**Comment by mattriverm** [comment, marked as answer]
I toyed around with rustdesk a few months ago, apparently I subscribed to this ticket :) 

The build script for scrap was not resolving the correct target name on M1.
To get by, I hardcoded
```rust
target = String::from("arm64-osx");
```

in libs/scrap/build.rs, after it has tried to resolve target on it's own.

If I recall correctly, I also had to place magnum-opus in libs and repeat the procedure.

Hope it helps you along
